src/util.py

```diff
@@ -0,0 +1,19 @@
+import pandas as pd
+import numpy as np
+from scipy.sparse import csr_matrix
+
+
+def compute_rating_matrix(df: pd.DataFrame):
+    N = len(df['user_id'].unique())
+    M = len(df['joke_id'].unique())
+
+    user_mapper = dict(zip(np.unique(df["user_id"]), list(range(N))))
+    movie_mapper = dict(zip(np.unique(df["joke_id"]), list(range(M))))
```

Comment:
Shouldn't it be joke_mapper instead of movie?

Reply:
I should ;)